task15/index.js

```diff
@@ -0,0 +1,92 @@
+const parent = document.getElementById("js-parent"); 
+const modalPlace = document.getElementById('js-modalPlace');
+
+const renderButtonForParent = () => {
+    const button = document.createElement('input');
+    modalPlace.style.display = "none"; 
+    button.id = "js-openBtn";
+    button.type = "button";
+    button.value = "click";
+    parent.appendChild(button);
+};
+renderButtonForParent();
+
+const openBtn = document.getElementById("js-openBtn");
+openBtn.addEventListener("click", (e) => {
+    modalPlace.className = "modal";
+    modalPlace.style.display = "block";
+    e.target.remove();
+    modalClose();
+});
+
+const modalClose = () => {
+    const closeBtn = document.getElementById("js-closeBtn");
+    closeBtn.addEventListener("click", () => {
+        requestValue();
+    });
+};
+
+const requestValue = () => {
+    const getName = document.getElementById("js-getName").value;
+    const getNumber = document.getElementById("js-getNumber").value;
+
+    if( getName.match(/\S/g) && getNumber){
+        console.log(getName);
+        console.log(getNumber);
+        modalPlace.style.display = "none";
+        init();
+        return
+    }
+        alert("入力してください");
+};
+
+const getFetchData = async () => {
+    const response = await fetch("https://myjson.dit.upm.es/api/bins/cvud");
+    const json = await response.json();
+    return json.data;
+};
+
+const addLoading = () => {
+    const loadingPlace = document.createElement("div");
+    const gif = document.createElement('img');
+    loadingPlace.id = "loadingPlace";
+    gif.src = "loading-circle.gif";
+    parent.appendChild(loadingPlace).appendChild(gif);
+};
+
+const removeLoading = () => document.getElementById("loadingPlace").remove();
+
+const renderListForParent = (value) => {
+    const ul = document.createElement('ul');
+    const fragment = document.createDocumentFragment();
+
+    value.forEach(d => {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.textContent = d.text;
+        a.href = d.a;
+        img.src = d.img;
+        img.alt = d.alt;
```

Comment:
IMO
I wondered what the "d" meant. Is it possible to change the name to something more understandable?

Reply:
The data in json was abbreviated to 'd'. My teacher once told me that it can be written that way, so I wrote it this way.
However, I changed it to 'value' because it was often pointed out throughout the assignment. What do you think of this change?
2b53035

Reply:
Oh I am sorry I did not know that thing:sob::sob: I will research about it later! Thank you for teaching me about it and fixing!!